src/QueuingEvented.ts

```diff
@@ -9,49 +9,38 @@ import Evented, { isGlobMatch, EventObject } from './Evented';
  *
  * @property maxEvents  The number of events to queue before old events are discarded. If zero (default), an unlimited number of events is queued.
  */
-export default class QueuingEvented extends Evented {
+class QueuingEvented<M extends {} = {}> extends Evented<M> {
 	private _queue: Map<string | symbol, EventObject[]>;
-	private _originalOn: (...args: any[]) => Handle;
 
 	maxEvents = 0;
 
 	constructor() {
 		super();
 
 		this._queue = new Map<string, EventObject[]>();
-		this._originalOn = this.on;
-		this.on = function (this: QueuingEvented, ...args: any[]): Handle {
-			let handle = this._originalOn(...args);
-
-			this.listenersMap.forEach((method, listenerType) => {
-				this._queue.forEach((events, queuedType) => {
-					if (isGlobMatch(listenerType, queuedType)) {
-						events.forEach((event) => this.emit(event));
-						this._queue.delete(queuedType);
-					}
-				});
-			});
-
-			return handle;
-		};
 	}
+}
 
-	emit<E extends EventObject>(event: E): void {
-		super.emit(event);
+(function (proto) {
+	QueuingEvented.prototype.emit = function emit(event: any): void {
+		proto.emit.call(this, event);
 
 		let hasMatch = false;
 
+		// @ts-ignore
 		this.listenersMap.forEach((method, type) => {
 			if (isGlobMatch(type, event.type)) {
 				hasMatch = true;
 			}
 		});
 
 		if (!hasMatch) {
+			// @ts-ignore
 			let queue = this._queue.get(event.type);
```

Comment:
What are we ignoring here?

Reply:
yeah, we really should comment why we are deciding to ignore something and would much prefer casting (and a comment)...  I have yet to see a valid use case yet for `// @ts-ignore` in our code base.

Reply:
These were because `_queue` and `listenersMap` weren't accessible outside of the class declaration of `QueuingEvented`. Since I've moved the functions into the class definition, these comments are no longer needed.